c/windows.c

```diff
@@ -52,17 +53,14 @@ void *S_ntdlsym(void *h, const char *s) {
     return (void *)GetProcAddress(h, s);
 }
 
-/* S_ntdlerror courtesy of Bob Burger, burgerrg@sagian.com */
-char *S_ntdlerror(void) {
-    static char s[80];
-    INT n;
-
-    n = FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM, NULL, GetLastError(),
-                      0, (LPTSTR)s, 80, NULL);
-    if (n == 0) return "unable to load library";
-  /* Strip trailing period, newline & return when present */
-    if (n >= 3 && s[n-3] == '.') s[n-3] = 0;
-    return s;
+/* Initial version of S_ntdlerror courtesy of Bob Burger, burgerrg@sagian.com
+ * Modifications by James-Adam Renquinha Henri, jarhmander@gmail.com */
+ptr S_ntdlerror(void) {
+    ptr ret;
+    char *s = s_ErrorStringImp(GetLastError(), "unable to load library");
+    ret = Sstring_utf8(s, -1);
```

Comment:
You might simply `ptr ret = ...`

Reply:
I thought that wasn't C89-compliant, but I guess I was wrong. In any case, this hunk is no longer relevant…